fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm

```diff
@@ -0,0 +1,481 @@
+	handed.force_wielded = upgraded_val(on_force, upgrade_level)
+	handed.force_unwielded = upgraded_val(base_force,upgrade_level)
+	force = handed.force_unwielded
+
+/obj/item/melee/trick_weapon/hunter_axe/update_icon_state()
+	icon_state = "[base_icon_state]0"
+	playsound(src,'sound/magic/clockwork/fellowship_armory.ogg',50)
+	return ..()
+
+/obj/item/melee/trick_weapon/hunter_axe/proc/on_wield(obj/item/source)
+	enabled = TRUE
+
+/obj/item/melee/trick_weapon/hunter_axe/proc/on_unwield(obj/item/source)
+	enabled = FALSE
+
+/obj/item/rabbit_eye
+	name = "Rabbit eye"
+	desc = "An item that resonates with trick weapons."
+	icon_state = "rabbit_eye"
+	icon = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons.dmi'
+
+/obj/item/rabbit_eye/proc/upgrade(obj/item/melee/trick_weapon/killer, mob/user)
+	if(killer.upgrade_level >= 3)
+		user.balloon_alert(user, "Already at maximum upgrade!")
+		return
+	if(killer.enabled)
+		user.balloon_alert(user, "Weapon must be in base form!")
+		return
+	SEND_SIGNAL(killer,WEAPON_UPGRADE)
+	killer.name = "[killer.base_name] +[killer.upgrade_level]"
+	to_chat(user, span_warning ("The [src] crumbles away..."))
+	qdel(src)
+
+/obj/item/gun/ballistic/revolver/hunter_revolver
+	name = "\improper Hunter's Revolver"
+	desc = "Does minimal damage but slows down the enemy."
+	icon_state = "revolver"
+	icon = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons.dmi'
+	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/bloodsilver
+	initial_caliber = CALIBER_BLOODSILVER
+
+/datum/movespeed_modifier/silver_bullet
+	movetypes = GROUND
+	multiplicative_slowdown = 4
+	flags = IGNORE_NOSLOW
+
+
+/obj/item/ammo_box/magazine/internal/cylinder/bloodsilver
+	name = "detective revolver cylinder"
+	ammo_type = /obj/item/ammo_casing/silver
+	caliber = CALIBER_BLOODSILVER
+	max_ammo = 2
+
+/obj/item/ammo_casing/silver
+	name = "Bloodsilver casing"
+	desc = "A Bloodsilver bullet casing."
+	icon_state = "bloodsilver"
+	icon = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons.dmi'
+	projectile_type = /obj/projectile/bullet/bloodsilver
+	caliber = CALIBER_BLOODSILVER
+
+
+/obj/projectile/bullet/bloodsilver
+	name = "Bloodsilver bullet"
+	damage = 3
+	ricochets_max = 4
+
+/obj/projectile/bullet/bloodsilver/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(!iscarbon(target))
+		return
+	var/mob/living/carbon/man = target
+	if(!man)
+		return
+	if(man.has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
+		return
+	if(!(IS_BLOODSUCKER(man)) && !(man.mind.has_antag_datum(/datum/antagonist/changeling)))
+		return
+	man.add_movespeed_modifier(/datum/movespeed_modifier/silver_bullet)
+	addtimer(CALLBACK(man, /mob/living/carbon.proc/remove_bloodsilver), 20 SECONDS)
+
+/mob/living/carbon/proc/remove_bloodsilver()
+	if (src.has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
```

Comment:
```suggestion
	if (has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
```
Why is this a carbon proc? why not just call ``remove_movespeed_modifier`` when you need?